### PR TITLE
Fix text within Pay button on checkout page

### PR DESCRIPTION
**Pull request by paulbrown1982**
Fix bug where the Pay £xx button on the checkout page did not change in line with the payment options.
We could probably refactor away the horrible options.js now!

cc @rtyley @tomverran

### CODE REVIEW

File: frontend/app/views/fragments/pricing/billingPeriodChoice.scala.html [hunk 1]
```diff
@@ -47,6 +47,9 @@
             generateChargeAmount: function() {
                 return formattedDiscountedCharge(checkoutForm.currency, this.amount, this.discount);
             },
+            generateDisplayAmount: function() {
```
Comment (rtyley) [inline frontend/app/views/fragments/pricing/billingPeriodChoice.scala.html]: There's starting to be a bit of repetition here, duplicated between monthly & annual billing. Perhaps some of it is removable?! (in a subsequent PR, not necessarily for this one)

**Comment by rtyley** [comment]
I'm not the best at javascript, but this looks :+1: to me.

File: frontend/app/views/fragments/pricing/billingPeriodChoice.scala.html (hunk 1 above)
Comment (paulbrown1982) [inline frontend/app/views/fragments/pricing/billingPeriodChoice.scala.html]: Yeah we should remove them from the choices array too. It was there to make the template cleaner, so it doesn't have to call back up the object model to find the correct generateDisplayAmount function, it's there on the choice it's looped over.

**Comment by rtyley** [comment]
cc @glockett

**Pull request merged**